app/views/posts/_post.html.erb

```diff
@@ -0,0 +1,22 @@
+<div class="border border-dark my-2">
+  <%= link_to user_post_path(post.author.id, post.id) do %>
+      <h3><%= post.title %></h3>
+        <% end %>
+      <p><%= post.text %></p>
+
+      <div>
+        <div>Comments: <%= post.comments_counter ? post.comments_counter : 0 %></div>
+        <div>Likes: <%= post.likes_counter ? post.likes_counter : 0 %></div>
+      </div>
+
+      <ul>
+        <% filtered_comments = comments.select{|a| a.post_id == post.id } %>
+        <% filtered_comments.each do |comment| %>
+              <li>
+                <%= comment.author.name %>
+                <%= comment.text %>
+              </li>
+            
+        <% end %>
+      </ul>
```

Comment:
- [OPTIONAL SUGGESTION] Following the above suggestion, we'll no longer need to filter the comments and this makes code leaner and our work easier. We can rewrite this part like this:

```ruby
      <ul>  
        <% comments.each do |comment| %>
          <li>
            <%= comment.author.name %>
            <%= comment.text %>
          </li>            
        <% end %>
      </ul>
```